### PR TITLE
bpo-25821: Fix inaccuracy in threading.enumerate/is_alive documentation

### PR DESCRIPTION
**Pull request by iritkatriel**
enumerate() returns a list of all active thread objects, which contains all threads for which is_alive is true but may also contain threads (like the main thread) for which is_alive is False. 

https://bugs.python.org/issue25821

### CODE REVIEW

**Comment by pitrou** [comment]
"active" is undefined. I would rather keep the corrent wording, but perhaps add that the main thread is _always_ there even when finished.

**Comment by iritkatriel** [comment]
I picked up active from 

```
.. function:: active_count()

   Return the number of :class:`Thread` objects currently alive.  The returned
   count is equal to the length of the list returned by :func:`.enumerate`.
```

**Comment by pitrou** [comment]
Hmm, well you see that "active" is a synonym of "alive" in that instance.

**Comment by iritkatriel** [comment]
These seems to be some distinction between "the thread object is alive"  and thread.is_alive()

**Comment by pitrou** [comment]
Perhaps the following will be clear (but more wordy):
```rst
   Return a list of all :class:`Thread` objects currently active.  The list
   includes daemonic threads, dummy thread objects created by
   :func:`current_thread`, and the main thread.  It excludes terminated threads
   and threads that have not yet been started. However, the main thread
   is always part of the result, even when terminated.
```

File: Lib/threading.py [hunk 1]
```diff
@@ -1109,7 +1109,7 @@ def is_alive(self):
 
         This method returns True just before the run() method starts until just
         after the run() method terminates. The module function enumerate()
-        returns a list of all alive threads.
+        returns a list that contains all alive threads.
```
Comment (taleinat) [inline Lib/threading.py]: I think the existing wording is fine here; not sure why this change is need?

**Review by taleinat — APPROVED** [review]
LGTM.

I think the change in threading.py is unnecessary, but harmless. @iritkatriel, let me know what you think about it, and then we can merge this.

File: Lib/threading.py (hunk 1 above)
Comment (iritkatriel) [reply Lib/threading.py]: That's because it may also contain a terminated main thread.

Comment (taleinat) [reply Lib/threading.py]: That makes the new wording correct, but possibly misleading.

The sentence about `enumerate()` seems somewhat out of place here; perhaps it could just be removed? Otherwise let's add something like "(and always the main thread)" at the end.

**Review by taleinat — CHANGES_REQUESTED** [review]

File: Lib/threading.py (hunk 1 above)
Comment (iritkatriel) [reply Lib/threading.py]: Perhaps just "see also .."?

**Comment by iritkatriel** [comment]
I have made the requested changes; please review again

**Review by taleinat — APPROVED** [review]

**Comment by iritkatriel** [comment]
Oh!
